packages/common/src/search/wellKnownCatalog.ts

```diff
@@ -0,0 +1,268 @@
+import { IUser } from "@esri/arcgis-rest-types";
+import { getFamilyTypes } from "../content";
+import { EntityType, IHubCatalog, IHubCollection } from "./types";
+
+export type WellKnownCatalog =
+  | "myContent"
+  | "myGroup"
+  | "favorites"
+  | "organization"
+  | "world";
+
+export type WellKnownCollection =
+  | "appAndMap"
+  | "content"
+  | "dataset"
+  | "document"
+  | "event"
+  | "feedback"
+  | "initiative"
+  | "project"
+  | "site"
+  | "solution"
+  | "template";
+
+/** This function returns a single IHubCatalog definition JSON object based on the
+ * name and entity type requested
+ * @param i18nScope Translation scope to be interpolated into the catalog
+ * @param name Name of the catalog requested
+ * @param entityType
+ * @param user Owner of the entity, optional but required for certain catalogs
+ * @param collectionNames A list of collection names, optional, if passed in,
+ * only those collections in the catalog will be returned
+ */
+export function getWellKnownCatalog(
+  i18nScope: string,
+  name: WellKnownCatalog,
+  entityType: EntityType,
+  user?: IUser,
+  collectionNames?: WellKnownCollection[]
```

Comment:
I'm worried that as we continue to expand this function out to other entity types, we might need other arguments. Already the two optionals can make it cumbersome if somebody doesn't have the user but wants certain collections returned. What if we created another interface like `IGetWellKnownCatalogOptions`? (name just suggested)

Reply:
Agree with @sonofflynn89 - group all the optional props into a single interface arg, which itself is optional.